### PR TITLE
README: Add fetch-upstream command

### PR DESCRIPTION
**Pull request by mounilKshah**
**What does this PR do?**
This PR makes it easier to clone the Zulip Terminal repository and setup upstream for the remote repository.

**Discussion:**
[zulip-terminal>Development Workflow](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/Development.20Workflow/near/1372760)

**Tested?**
- [x] Manually
- [x] Existing tests (adapted, if necessary)
- [x] Passed linting & tests (each commit)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,16 +225,26 @@ You can of course browse the source on GitHub & in the source tree you download,
 
 Zulip Terminal uses [urwid](http://urwid.org/) to render the UI components in terminal. Urwid is an awesome library through which you can render a decent terminal UI just using python. [Urwid's Tutorial](http://urwid.org/tutorial/index.html) is a great place to start for new contributors.
 
+### Getting Zulip Terminal code and connecting it to upstream
+
+First, fork the `zulip/zulip-terminal` repository ([see how](https://docs.github.com/en/get-started/quickstart/fork-a-repo)) and then clone your forked repository locally (the following will place the repository in the current directory):
+```
+$ git clone --config pull.rebase git@github.com:YOUR_USERNAME/zulip-terminal.git
```
Comment (neiljp) [inline README.md:232]: Does other documentation like this describe `YOUR_USERNAME`?

File: README.md [hunk 2]
```diff
@@ -225,16 +225,26 @@ You can of course browse the source on GitHub & in the source tree you download,
...
 
 Various options are available; we are exploring the benefits of each and would appreciate feedback on which you use or feel works best.
 
 Note that the tools used in each case are typically the same, but are called in different ways.
 
-With any option, you first need to clone the zulip/zulip-terminal repository locally (the following will place the repository in the current directory):
-```
-$ git clone git@github.com:zulip/zulip-terminal.git
-```
 The following commands should be run in the repository directory, which can be achieved with `cd zulip-terminal`.
```
Comment (neiljp) [inline README.md:248]: Maybe adjust this line to refer to the 'repository directory from the previous section' or similar?

**Review by neiljp — COMMENTED** [review]
@mounilKshah This reads well and is a cleaner change than the first draft we discussed :+1:

File: README.md (hunk 1 above)
Comment (mounilKshah) [reply README.md:232]: Only [Zulip](https://zulip.readthedocs.io/en/latest/git/cloning.html#step-1b-clone-to-your-machine) has it as `YOURUSERNAME`. Zulip-mobile and desktop don't have instructions regarding cloning the _forked_ repository.

**Comment by neiljp** [comment]
@mounilKshah Thanks for this - as per discussion, this is now merged with the follow-up commit I put together :tada:

**Pull request closed without merging**